lib/artsy-eventservice/artsy/event_service/publisher.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+module Artsy
+  module EventService
+    class Publisher
+      include RabbitMQConnection
+
+      def self.publish(topic:, event:)
+        new.post_event(topic: topic, event: event)
+      end
+
+      def post_event(topic:, event:)
+        raise 'Event missing topic or verb.' if event.verb.to_s.empty? || topic.to_s.empty?
+        connect_to_rabbit do |conn|
```

Comment:
So if we create the connection as `@conn` when the EventService class is instantiated, then create a method `with_channel` instead:
```
def with_channel
  channel = @conn.create_channel
  yield channel
  channel.close
end
```

And use it like:
```
with_channel do |channel|
  exchange = channel.topic(topic, durable: true)
  ...
end
```